zanata-war/src/main/java/org/zanata/action/LanguageJoinAction.java

```diff
@@ -97,65 +92,164 @@
     @Setter
     private String message;
 
+    @In
+    private RequestService requestServiceImpl;
+
+    @In
+    private ZanataIdentity identity;
+
+    @In
+    private LanguageTeamService languageTeamServiceImpl;
+
     @In(value = ZanataJpaIdentityStore.AUTHENTICATED_USER, required = false)
     private HAccount authenticatedAccount;
 
-    public boolean hasSelectedRole() {
-        return requestAsTranslator || requestAsReviewer || requestAsCoordinator;
+    public String getMyRequestedRole() {
```

Comment:
What are we doing with this string? It seems a bit non-typesafe. 

Even if it is for immediate display to the user, we should consider internationalisation of role names (eg the `Messages` class).